pkgs/development/interpreters/python/hooks/python-relax-deps-hook.sh

```diff
@@ -39,16 +39,20 @@
 #
 # Currently unsupported: URL specs (foo @ https://example.com/a.zip).
 
+# shellcheck source=pkgs/stdenv/generic/setup.sh
+. /dev/null
```

Comment:
I think that it (of course) does nothing at runtime, but since ShellCheck can’t reliably know what e.g. `. $dynamicVar/foo.sh` will do, it has a directive to tell it explicitly what to treat you as having sourced, so that it knows about all the things in the `stdenv` setup.

I actually quite like this as a dependency declaration and wouldn’t mind just putting it in all the hook files, but perhaps we could reasonably do it in a central place. It would be lovely to get shell hooks to condition their build on ShellCheck results like `writeShellApplication`…

Reply:
ShellCheck's `source` directive specifies the path to the sourced script (absolute or relative to where ShellCheck is executed.). The trick of `. /dev/null` is that it does nothing at run time (`/dev/null behaves like an empty file when reading) but tells ShellScript that `setup.sh` will be sourced. I assume that ShellCheck will always be executed at the root of Nixpkgs.

An alternative is the `source-path` directive. It also specifies the path to the sourced script, but when specified with a relative path, it is considered relative to the `basedir` of the current directory.

Here is the documentation of ShellCheck directives:
https://www.shellcheck.net/wiki/Directive

Reply:
I think we could plausibly make `makeSetupHook` run ShellCheck and make the `stdenv` bootstrap override it. OTOH I don’t know if we’d want to pull in that much Haskell closure, so maybe CI is best.